tests/framework/e2e/downgrade.go

```diff
@@ -117,6 +131,14 @@ func DowngradeUpgradeMembersByID(t *testing.T, lg *zap.Logger, clus *EtcdProcess
 			return err
 		}
 	}
+
+	if opString == "downgrading" && len(membersToChange) == len(clus.Procs) {
+		lg.Info("Waiting for downgrade completion log line")
+		leader := clus.WaitLeader(t)
+		_, err := clus.Procs[leader].Logs().ExpectWithContext(context.TODO(), expect.ExpectedResponse{Value: "the cluster has been downgraded"})
```

Comment:
@ahrtr I tried to implement [this comment](https://github.com/etcd-io/etcd/pull/19399/files#r1952973254) by monitoring the log for the leader. But I am not sure if this is the best way to go about it (e.g., the leader is re-elected during the time I am monitoring the log, etc.)